jplephem/spk.py

```diff
@@ -228,9 +228,13 @@ def generate(self, tdb, tdb2):
 
         if (index < 0).any() or (index > n).any():
             final_epoch = init + intlen * n
-            raise InvalidTimestampError('segment only covers dates %.1f through %.1f' % (init, final_epoch),
-                                        min_timestamp=init,
-                                        max_timestamp=final_epoch)
+            raise OutOfRangeTimestampError(
+                'segment only covers dates %.1f through %.1f' % (init,
+                                                                 final_epoch),
+                min_timestamp=init,
+                max_timestamp=final_epoch,
+                out_of_range_times=[True if i < 0 or i > n else False
```

Comment:
This is correct (yay!), but requires Python to loop over all million elements of the vector if a million-element NumPy vector of times was provided. I think you can use the same inequalities as in the `if` statement — could you try this and see how if it works or not?

`(index < 0) | (index > n)`

Reply:
Sorry, I'm not really familiar with NumPy and to be fair, I don't really know how I should proceed to test it manually (that's why I was giving hope on the unit tests until here)…
Can you guide me on this? :/

Reply:
In that case I'll go ahead and merge, and add the test myself, then comment here so you can take a look and know how to add a unit test next time.